android/engine/src/main/java/org/smartregister/fhircore/engine/domain/model/FhirResourceConfigs.kt

```diff
@@ -90,7 +91,55 @@ data class ResourceConfig(
   val configRules: @RawValue List<RuleConfig>? = null,
   val planDefinitions: List<String>? = null,
   val attributesToUpdate: List<KeyValueConfig>? = emptyList()
-) : Parcelable, java.io.Serializable
+) : Parcelable, java.io.Serializable {
+  fun interpolate(computedValuesMap: Map<String, Any>) =
+    this.copy(
+      dataQueries = interpolateDataQueries(dataQueries = dataQueries, computedValuesMap),
+      relatedResources =
+        relatedResources.map {
+          it.copy(
+            dataQueries = interpolateDataQueries(dataQueries = dataQueries, computedValuesMap)
+          )
+        }
+    )
+
+  private fun interpolateDataQueries(
```

Comment:
Can we use the [FilterCriterionConfig.computedRule](https://github.com/opensrp/fhircore/blob/main/android/engine/src/main/java/org/smartregister/fhircore/engine/domain/model/DataQueryConfigs.kt#L43) instead of the value? The reason is, the value here is dynamically typed. The `computedRule` refers to the name of the rule that is executed to provide dynamic value for the dataQuery. interpolation only works with String types. I think that is why the `QuantityFilterCriterionConfig`, `DateFilterCriterionConfig`, and `NumberFilterCriterionConfig` were left out in this `when` statement (from a configs perspective we may have use cases that need dynamic content for these criteria).

Reply:
I missed this before. This will be in line with what [Search.filterBy](https://github.com/opensrp/fhircore/blob/79fd1f156534398ce797fa0974ccee05b5899fba/android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/SearchExtension.kt#LL58C15-L58C15) function expects.